Add python 3.10 build

calum-chamberlain opened the pull request:
…nda-forge-pinning 2022.12.27.19.59.10

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
~~* [ ] Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

calum-chamberlain committed:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.22.1, and conda-forge-pinning 2022.12.27.19.59.10

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

calum-chamberlain committed:
bump build no

calum-chamberlain committed:
Do not use static linking - .a files not provided beyond 2021

calum-chamberlain commented:
This looks good to me - I added python 3.10 and removed python 3.7 from the builds. I had to stop static linking because fftw doesn't provide .a files anymore. This is probably for the best and I can't seem to find the logic for using static linking in the first place (I'm sure there was one...).

I tested this locally with an env created from a build_local package on linux and it all worked well as well.
